modules/core/src/main/scala/doobie/util/analysis.scala

```diff
@@ -20,9 +20,28 @@ object analysis {
 
   /** Metadata for the JDBC end of a column/parameter mapping. */
   final case class ColumnMeta(jdbcType: JdbcType, vendorTypeName: String, nullability: Nullability, name: String)
+  object ColumnMeta {
+    def apply(jdbcType: JdbcType, vendorTypeName: String, nullability: Nullability, name: String): ColumnMeta = {
+      new ColumnMeta(tweakJdbcType(jdbcType, vendorTypeName), vendorTypeName, nullability, name)
+    }
+  }
 
   /** Metadata for the JDBC end of a column/parameter mapping. */
   final case class ParameterMeta(jdbcType: JdbcType, vendorTypeName: String, nullability: Nullability, mode: ParameterMode)
+  object ParameterMeta {
+    def apply(jdbcType: JdbcType, vendorTypeName: String, nullability: Nullability, mode: ParameterMode): ParameterMeta = {
+      new ParameterMeta(tweakJdbcType(jdbcType, vendorTypeName), vendorTypeName, nullability, mode)
+    }
+  }
+
+  private def tweakJdbcType(jdbcType: JdbcType, vendorTypeName: String) = jdbcType match {
+    // the Postgres driver does not return *WithTimezone types but they are pretty much required for proper analysis
+    // https://github.com/pgjdbc/pgjdbc/issues/2485
+    // https://github.com/pgjdbc/pgjdbc/issues/1766
+    case JdbcType.Time if vendorTypeName.compareToIgnoreCase("timetz") == 0 => JdbcType.TimeWithTimezone
+    case JdbcType.Timestamp if vendorTypeName.compareToIgnoreCase("timestamptz") == 0 => JdbcType.TimestampWithTimezone
```

Comment:
or `"TIMESTAMP WITH TIMEZONE"`?

Reply:
Are there JDBC drivers that return a type of `Timestamp` but a vendor type name of `TIMESTAMP WITH TIMEZONE`?

Reply:
What I meant was like this
```scala
    case JdbcType.Timestamp if vendorTypeName.compareToIgnoreCase("timestamptz") == 0 || vendorTypeName.compareToIgnoreCase("TIMESTAMP WITH TIMEZONE") == 0 => ...
```

Reply:
Is `case JdbcType.Timestamp if vendorTypeName.compareToIgnoreCase("TIMESTAMP WITH TIMEZONE") == 0` ever true though?

Postgres returns `timestamptz`, h2 returns `JdbcType.TimestampWithTimezone`, mySQL doesn't have a timestamp with timezone column type

Reply:
Fair enough :+1: